hostfactory/host_provider/src/allocation_strategy.py

```diff
@@ -152,7 +152,9 @@ def allocate_slots(self, requested_slot_count, template_id, vm_types):
                      template_id, self.auto_scaling_strategy)
         # Filter out vmTypes that have no available capacity
         filtered_vm_types = self.filter_available_vmTypes(vm_types)
-
+        if len(filtered_vm_types) == 0:
+            self.logger.warning("No available VM types found - cannot allocate nodes")
+            return False
```

Comment:
instead of False, this should return []

Reply:
Yeah I can do empty list but ultimately we do return False in add_nodes in cluster.py